doc/guides/west/build-flash-debug.rst

```diff
@@ -19,6 +22,71 @@ build system commands with the same names directly delegate to West.
    .. contents::
       :local:
 
+.. _west-building:
+
+Building: ``west build``
+************************
+
+.. tip:: Run ``west build -h`` for additional help.
+
+The ``build`` command allows you to build any source tree from any directory
+in your file system, placing the build results in a folder of your choice.
+
+In its simplest form, the command can be run by navigating to the root folder
+(i.e. the folder containing a file:`CMakeLists.txt` file) of the Zephyr
+application of your choice and running::
+
+  west build -b <BOARD>
+
+Where ``<BOARD>`` is the name of the board you want to build for. This is
+exactly the same name you would supply to CMake if you were to invoke it with:
+``cmake -DBOARD=<BOARD>``.
+A build folder named :file:`./build` (default name) will be created and the
+build output will be placed in it.
+
+To specify the build directory, use ``--build-dir`` (or ``-d``)::
+
+  west build -b <BOARD> --build-dir path/to/build/directory
+
+Since the build directory defaults to :file:`./build`, if you do not specify
+a build directory but a folder named `./build` is present that will be used,
+allowing you to incrementally build from outside the file:`build` folder with
+no additional parameters.
+
+.. note::
+  The ``-b <BOARD>`` parameter is only required when there is no CMake cache
+  present at all, usually when you are building from scratch or creating a
+  brand new build directory. After that you can rebuild (even with additional
+  parameters) without having to specify the board again.
+
+To specify the source directory, use ``--source-dir`` (or ``-s``)::
+
+  west build -b <BOARD> --source-dir path/to/source/directory
+
+Additionally you can specify the target to be specified to the build system
+using the ``--target`` (or ``-t``) option::
+
+  west build -t clean
+
+You can list all targets with ``ninja help`` inside the build folder.
+
+Finally, you can also fine-tune the CMake invocation that happens under the
+hood by using the ``--cmake`` (or ``-c``) option to force CMake to be run
+and/or supplying additional parameters to CMake itself::
+
```

Comment:
It's less confusing to consistently put the description before the command line example as you've been doing, but here the description is after, so change to:
```
This command forces CMake to re-run and update its cache::

  west build -c

This instructs CMake to use Make instead of ninja (the default generator)::

   west build -b <BOARD> -- -G"Unix Makefiles"

This passes the ``OVERLAY_CONFIG`` definition to the build system so that your
file:`file.conf` is included in Kconfig::

   west build -b <BOARD> -- -DOVERLAY_CONFIG=file.conf
 
```